### ISSUE TITLE
Cyr

### ISSUE THREAD

**Issue by bgruening** [galaxyproteomics/tools-galaxyp#108]

**Comment by iracooke** [comment]
@bgruening I've been working on a homebrew recipe.  Works on OSX and I'm hoping to get it accepted.  I have no experience with conda but maybe the homebrew recipe could be adapted?

https://github.com/Homebrew/homebrew-science/pull/4555

**Comment by bgruening** [comment]
@iracooke thanks I tried this. But conda is way more general and least how we do it. We want to have every dependency defined and we want to run in in environments where perl is not under /usr/bin/perl etc ... maybe we can sit together sometime during the next two weeks and try to fix this?

Thanks!

**Comment by iracooke** [comment]
@bgruening I didn't mean that the galaxy tools should use homebrew but the source code to that recipe shows a modified TPP I made to help with packaging  https://github.com/iracooke/tpp  . Maybe that is helpful to fix the conda build.  I have no idea what the problems are currently with conda.

**Comment by iracooke** [comment]
@bgruening Yes.  I would like to try and help fix this.  Maybe something for this weekend.  Looking fwd to seeing you in Melbourne 👍

**Comment by Stortebecker** [comment]
Hejhej.
Do you still think that the TPP is needed? I have got the impression that it is not really state-of-the-art any more. But I might be wrong. In my experience, the TPP has good ideas and some really good tools, but the implementation of most tools is really bad. After so many years, most questions on their list still regard problems with the installation. So I gave up on the TPP. However, making the tools run in Galaxy would solve a lot of practical problems with the TPP framework.

**Comment by iracooke** [comment]
Yes indeed.  Installation of the TPP is a monumental pain.  The tools are good in some respects though.  As a general rule they are written to be fast and they support a very wide variety of search engines and workflows, not to mention the many statistical ideas they encapsulate.  Since they are widely used they are also fairly well tested and validated. There are certainly alternatives out there but many of those have their own problems, or only replace a subset of what the TPP does.  So despite it's problems I think the TPP is still relevant.

Having said that I'm afraid I haven't got the time to implement a conda recipe.  I did get a homebrew recipe accepted into homebrew science though.  If someone could adapt that to linux it would be awesome.

**Comment by Stortebecker** [comment]
@iracooke Sorry, but I do not understand enough of programming to do the job. I will try to gently push my colleagues, but they are also *really* buisy.

Speaking of the TPP: another drawback is the bad documentation, or do you know good tutorials, especially on the more complex statistics stuff? I did not use half of it, because it was so tedious to find out what everything is really doing and if it's worth the effort.

**Comment by iracooke** [comment]
If you don't have a specific need for the TPP its probably a good idea to just not worry about it.  Getting the tools working well does indeed take a lot of work.  I have made a set of ruby wrappers for some of the tools (PROTK ruby gem) but there are many tools I have not wrapped and unless you use MacOS you will still have the installation problem.
